blocks/cards/cards.js

```diff
@@ -36,38 +72,23 @@ export default async function decorate(block) {
   const ul = document.createElement('ul');
   [...block.children].forEach((row) => {
     const li = document.createElement('li');
-    li.innerHTML = row.innerHTML;
-    [...li.children].forEach((div) => {
-      if (div.children.length === 1 && div.querySelector('picture')) div.className = 'cards-card-image';
-      else {
-        div.className = 'cards-card-body';
-        const bubble = div.querySelector('u');
-        if (bubble) {
-          bubble.className = 'cards-card-bubble';
-          bubble.closest('p').className = 'cards-card-bubble-wrapper';
-        }
-        const subtitle = div.querySelector('h2 + p > strong');
-        if (subtitle && subtitle.parentNode.textContent === subtitle.textContent) {
-          const title = div.querySelector('h2');
-          const titleWrapper = document.createElement('div');
-          titleWrapper.className = 'cards-card-title';
-          titleWrapper.append(title.cloneNode(true), subtitle.parentNode);
-          title.replaceWith(titleWrapper);
-        }
-        const country = div.querySelector('.icon[class*=icon-flag-]');
-        if (country) {
-          country.closest('p').classList.add('cards-card-country');
-          div.classList.add('cards-card-country-wrapper');
-        }
-        const list = div.querySelector('ul, ol');
-        if (list) {
-          const links = document.createElement('div');
-          links.className = 'cards-card-links';
-          links.append(list);
-          div.after(links);
-        }
-      }
-    });
+    // wrap main card element in an anchor if card has button
+    const button = row.querySelector('.button');
+    if (button && button.href) {
+      const title = row.querySelector('h2');
+      const a = document.createElement('a');
+      a.href = button.href;
+      a.title = title ? title.textContent : button.title;
+      a.setAttribute('aria-label', title ? title.textContent : button.title);
+      // hide duplicate link from screenreaders
+      button.setAttribute('tabindex', -1);
```

Comment:
Does the nested `a` need to be removed/transformed for accessibility? Or is this sufficient to hide the duplication from screen readers?